src/lib.rs

```diff
@@ -278,28 +295,31 @@ pub mod pallet {
 		#[pallet::weight(T::WeightInfo::submit_call(call.encode().len() as u32))]
 		pub fn submit_call(
 			origin: OriginFor<T>,
-			supersig_id: T::AccountId,
+			supersig_account: T::AccountId,
 			call: Box<<T as pallet::Config>::Call>,
 		) -> DispatchResult {
 			let who = ensure_signed(origin)?;
+			let supersig_id = Self::get_supersig_id_from_account(&supersig_account)?;
 
-			let supersig_index = Self::get_supersig_index_from_id(&supersig_id)?;
 			let data = call.encode();
 
-			let deposit =
-				<BalanceOf<T>>::from(data.len() as u32).saturating_mul(T::PricePerByte::get());
+			// Incentive to remove proposal that won't be accepted
+			let deposit = Self::compute_deposit(data.len())?;
 			T::Currency::reserve(&who, deposit)?;
```

Comment:
This could fail after a successfull storage mutation. The extrinsic should be marked `#[transactional]`

Reply:
Please do